# **MÓDULO 13**
# Projeto - Fundamentos da Descoberta de Dados

Nesse projeto trabalharemos com a base de dados de produtos de um supermercado do Chile.
A ideia é que vocês apliquem os conceitos estatísticos vistos no último módulo, mais os conceitos de visualizações de dados através de gráficos e finalizem publicando no seu github!

In [3]:
import pandas as pd
import matplotlib.pyplot as plt
import plotly.express as px

Faça a leitura dos dados do arquivo CSV:


Altere o código abaixo de acordo com seu diretório.

In [11]:
df = pd.read_csv('/usr/local/lib/python3.11/dist-packages/pandas/io/MODULO7_PROJETOFINAL_BASE_SUPERMERCADO - MODULO7_PROJETOFINAL_BASE_SUPERMERCADO (1).csv.csv', delimiter=',')

df.head(10)

title       Marca  Preco_Normal  \
0              Pack 12 un, Leche extra proteína 1 L  Loncoleche         19788   
1   Pack 12 un, Leche chocolate receta original 1 L     Soprole         18228   
2    Pack 12 un, Leche semidescremada chocolate 1 L     Soprole         18228   
3     Pack 12 un, Leche semidescremada frutilla 1 L     Soprole         18228   
4       Pack 12 un, Leche sin lactosa chocolate 1 L  Loncoleche         17988   
5        Pack 12 un, Leche sin lactosa frutilla 1 L  Loncoleche         17988   
6  Pack 12 un, Leche saborizada light chocolate 1 L  Loncoleche         17988   
7         Pack 12 un, Leche saborizada frutilla 1 L       Colun         17388   
8         Pack 12 un, Leche saborizada vainilla 1 L       Colun         17388   
9           Pack 12 un, Leche saborizada manjar 1 L       Colun         17388   

   Preco_Desconto  Preco_Anterior  Desconto Categoria  
0               0               0         0   lacteos  
1               0               0         0   lacteos  
2               0               0         0   lacteos  
3               0               0         0   lacteos  
4               0               0         0   lacteos  
5               0               0         0   lacteos  
6               0               0         0   lacteos  
7               0               0         0   lacteos  
8               0               0         0   lacteos  
9               0               0         0   lacteos

Os campos do nosso dataframe são:

**Title:** Nome do produto.


**Marca:** A marca do produto.


**Preco_Normal:** O preço em que o produto costuma ser vendido quando não há desconto.


**Preco_Desconto:** O preço vendido após o desconto ser aplicado.


**Preco_Anterior:** Preço em que era comercializado o produto antes do desconto aplicado.


**Desconto:** Total de desconto aplicado.






As colunas que aparecem com valores 0 são para os produtos onde não tivemos descontos aplicados.


As categorias estão em espanhol!

# 1 - Traga a média e a mediana dos preços - coluna Preco_Normal - por categoria de produto.
# Identifique as categorias que parecem ter um valor de média abaixo ou acima da mediana.

In [12]:
#Seu código aqui para a média
df.groupby('Categoria')['Preco_Normal'].mean().reset_index()

Categoria  Preco_Normal
0  belleza-y-cuidado-personal   1783.556485
1          comidas-preparadas   3095.043478
2                  congelados   2108.042553
3                      frutas   1724.473684
4        instantaneos-y-sopas    765.491228
5                     lacteos   2385.219239
6                    verduras   1343.296875

In [13]:
#Seu código aqui para a mediana
df.groupby('Categoria')['Preco_Normal'].median().reset_index()

Categoria  Preco_Normal
0  belleza-y-cuidado-personal        1569.0
1          comidas-preparadas        3290.0
2                  congelados        1519.0
3                      frutas        1195.0
4        instantaneos-y-sopas         439.0
5                     lacteos         989.0
6                    verduras        1180.0

Digite aqui as categorias:

Lacteos, Frutas e Congelados.

# 2 - Traga o desvio padrão por categoria de produto.
# Qual o comportamento da média e mediana nas categorias com maior desvio?

In [14]:
#Seu código aqui
df.groupby('Categoria')['Preco_Normal'].std().reset_index()

Categoria  Preco_Normal
0  belleza-y-cuidado-personal   2210.041719
1          comidas-preparadas   2019.911428
2                  congelados   2111.539896
3                      frutas   1639.151114
4        instantaneos-y-sopas   1170.232869
5                     lacteos   3925.816164
6                    verduras   1012.699625

Digite nesse campo o comportamento que você identificou.

A categoria com maior Desvio Padrão é a Lacteos com um desvio de 3925.816164, o que sugere uma grande variabilidade nos preços, possivelmente devido a outliers muito elevados. A sua média (2385.21) é muito maior que a mediana (989.0), indicando uma distribuição assimétrica à direita, tendo em mente uma distribuição normal.

# 3 - Plot um boxplot da distribuição do Preco_Normal para a categoria que você identificou que tem o maior desvio padrão. Como é a distribuição desses dados segundo o boxplot? Você identifica muitos outliers?

Dica: Para trazer apenas os dados da categoria que você deseja você pode usar o df.loc[df['Categoria'] == 'CATEGORIA ESCOLHIDA'

In [15]:
# Filtrando os dados para a categoria "lacteos"
df_lacteos = df[df['Categoria'] == 'lacteos']

# Criando o boxplot
fig = px.box(df_lacteos, y='Preco_Normal', title="Distribuição de Preço - Lácteos", points="all")

# Exibir o gráfico
fig.show()

# 4 - Plote um gráfico de barras onde temos a média de descontos por categoria.

In [16]:
# Calculando a média de desconto por categoria
df_mean = df.groupby('Categoria')['Desconto'].mean().reset_index().sort_values(by='Desconto', ascending=False)

# Criando o gráfico de barras
fig = px.bar(df_mean, x='Categoria', y='Desconto',
             title="Média de Descontos por Categoria",
             labels={'Desconto': 'Média de Desconto'},
             color='Categoria')

# Removendo a legenda
fig.update_layout(showlegend=False)

# Exibir o gráfico
fig.show()


# 5 - Plote um gráfico de mapa interativo agrupando os dados por categoria, marca e trazendo a média de desconto.

In [ ]:
# Fazendo o agrupamento dos dados por Categoria e Marca, calculando a média do desconto para cada um.
df_group = df.groupby(['Categoria', 'Marca'])['Desconto'].mean().reset_index()

# Plotando o TreeMap
fig = px.treemap(df_group,
                 path=['Categoria', 'Marca'],
                 values='Desconto',
                 title='Desconto por Categoria e Marca',
                 width=1100,  # Largura do gráfico
                 height=600)  # Altura do gráfico

# Exibindo o gráfico
fig.show()